packages/plugin-react-native-client-sync/client-sync.js

```diff
@@ -1,4 +1,5 @@
 const { DeviceEventEmitter, NativeEventEmitter, NativeModules, Platform } = require('react-native')
+const makeSafe = require('@bugsnag/delivery-react-native/make-safe')
```

Comment:
I don't think it makes sense for this to import something from `delivery-react-native`

Should `make-safe` be a new plugin? We could put it in core, but that would increase the browser bundle size, so maybe it could go in `@bugsnag/react-native`?

Reply:
It'd be good to give it a better name now it's being used in multiple places as well; maybe something like "resolve recursive references" (or something less alliterative!)

Reply:
I mostly agree with moving it out of `delivery`, but moving it into `@bugsnag/react-native` leads us to a circular dependency. Given how tied to the react-native bridge this logic is, it should definitely be kept away from `core`.

Also agree on changing the name, maybe `bridgeSafe`?

Alternatively it needs to become more generic and move completely to its own module?

Reply:
> Given how tied to the react-native bridge this logic is&hellip;

I'm not sure it is tied to the react-native bridge; it just resolves recursive references in any JS value, it isn't RN specific AFAICT

Reply:
In the abstract, yes. On the other hand it also converts `Date` objects to strings, changes how `Error` objects are represented, etc. while those are probably useful behaviours everywhere, they're also baked into the logic right now.